src/ray/util/event.h

```diff
@@ -164,6 +164,16 @@ class RayEvent {
     return *this;
   }
 
+  RayEvent &WithField(const std::string &key, const std::string &value) {
+    custom_fields_[key] = value;
+    return *this;
+  }
+  RayEvent &WithFields(
+      const std::unordered_map<std::string, std::string> &custom_fields) {
+    custom_fields_ = custom_fields;
```

Comment:
hmm, should we merge the values into custom_fields other than replacement?

Reply:
Yeah I think this is a better UX. Ideally, we should set the custom fields once in the initialization (per component) and add withFields for each event.